src/server_manager/web_app/gcp_server.ts

```diff
@@ -0,0 +1,133 @@
+  private static readonly GUEST_ATTRIBUTES_POLLING_INTERVAL_MS = 5 * 1000;
+
+  private readonly gcpHost: GcpHost;
+  private installState: InstallState = InstallState.UNKNOWN;
+
+  // TODO: Consider passing the refreshToken instead of the client.
+  constructor(
+      id: string, private projectId: string, private instance: gcp_api.Instance,
+      private apiClient: gcp_api.RestApiClient) {
+    super(id);
+    this.gcpHost = new GcpHost(projectId, instance, apiClient);
+  }
+
+  getHost(): ManagedServerHost {
+    return this.gcpHost;
+  }
+
+  isInstallCompleted(): boolean {
+    return this.installState !== InstallState.UNKNOWN;
+  }
+
+  async waitOnInstall(): Promise<void> {
+    while (this.installState === InstallState.UNKNOWN) {
+      const outlineGuestAttributes = await this.getOutlineGuestAttributes(
+          this.projectId, this.instance.id, this.instance.zone);
+      if (outlineGuestAttributes.has('apiUrl') && outlineGuestAttributes.has('certSha256')) {
+        const certSha256 = outlineGuestAttributes.get('certSha256');
+        const apiUrl = outlineGuestAttributes.get('apiUrl');
+        trustCertificate(btoa(certSha256));
+        this.setManagementApiUrl(apiUrl);
+        this.installState = InstallState.SUCCESS;
+      } else if (outlineGuestAttributes.has('install-error')) {
+        this.installState = InstallState.ERROR;
+        throw new errors.ServerInstallFailedError();
+      }
+
+      await sleep(GcpServer.GUEST_ATTRIBUTES_POLLING_INTERVAL_MS);
+    }
+
+    // TODO: Handle user clicking cancel and deleting server.
+  }
+
+  private async getOutlineGuestAttributes(projectId: string, instanceId: string, zone: string):
+      Promise<Map<string, string>> {
+    const result = new Map<string, string>();
+    const guestAttributes =
+        await this.apiClient.getGuestAttributes(projectId, instanceId, zone, 'outline/');
+    // console.log(`Guest attributes: ${JSON.stringify(guestAttributes)}`);
+    const attributes = guestAttributes?.queryValue?.items;
+    if (attributes) {
+      const apiUrlAttr = attributes.find((attribute) => {
+        return attribute.key === 'apiUrl';
+      });
+      const certSha256Attr = attributes.find((attribute) => {
+        return attribute.key === 'certSha256';
+      });
+      const installErrorAttr = attributes.find((attribute) => {
+        return attribute.key === 'install-error';
+      });
+
+      if (apiUrlAttr) {
+        result.set('apiUrl', apiUrlAttr.value);
+      }
+      if (certSha256Attr) {
+        result.set('certSha256', certSha256Attr.value);
+      }
+      if (installErrorAttr) {
+        result.set('install-error', installErrorAttr.value);
+      }
+    }
+    return result;
+  }
+}
+
+class GcpHost implements server.ManagedServerHost {
+  constructor(
+      private projectId: string, private instance: gcp_api.Instance,
+      private apiClient: gcp_api.RestApiClient) {}
+
+  async delete(): Promise<void> {
```

Comment:
This should mark a server being installed as deleted, like we do for DO.

Reply:
Thanks, I saw that comment in #864 and addressed it there, but will backfill.

I forgot to add a TODO to clean up the other resources we create for the server (added now)

Reply:
Ugh, this makes me wonder if we would be better off having a project per server. Then I guess we could simply delete the project? Or would we still need to delete each resource before deleting the project?
Probably too late to change now though.

We will need to handle a partially deleted server. We should still show it in the UI, but broken, so the user has the chance to try to delete it again.

This means that you should go over the deletion steps again, making sure it doesn't fail on the steps that already completed before.